app/models/batch_invitation_user.rb

```diff
@@ -34,21 +34,23 @@ def humanized_outcome
   end
 
 private
+  def new_supported_permissions_for_user(supported_permission_ids)
+    if user = User.find_by_email(self.email.to_s.downcase.strip)
+      supported_permission_ids | user.supported_permission_ids
+    else
+      supported_permission_ids
+    end
+  end
+
   def invite_user_with_attributes(sanitised_attributes, inviting_user)
-    if User.find_by_email(self.email)
-      self.update_column(:outcome, "skipped")
+    user = User.invite!(sanitised_attributes, inviting_user)
```

Comment:
I wonder, does `invite!` behave the same as rails' `save!`, `update!` etc. As-in does it raise exceptions on failure. If so we can drop the conditional here.